guide/creating-your-bot/parsing-mention-arguments.md

```diff
@@ -0,0 +1,219 @@
+		!avatar @User
+	</discord-message>
+	<discord-message author="User" avatar="blue" :bot="true">
+		User's avatar: https://cdn.discordapp.com/avatars/208330347295932416/e2456a4bc63ec0215334434ade7fa45c.png
+	</discord-message>
+</discord-messages>
+
+Lo and behold, it does work.
+
+So now, instead of using `message.mentions` you can use your new, fantastic function.
+This will allow you to add proper checks for all your args, so that you can tell when a command was used correctly and when it was used incorrectly.
+
+But this does not mark the end of the page. If you feel adventurous you can read on and learn how to use Regular Expressions to easily convert a mention into a user object in just two lines.
+
+### Using Regular Expressions
+
+Previously you learn how to use rudimentary string related functions to turn the special mention syntax Discord uses into a proper Discord.js User object.
+But using Regular Expressions (aka "RegEx" or "RegExp"), you can condense all that logic into a single line! Crazy, right?
+
+If you have never worked with Regular Expressions before, this might seem daunting. But in fact, you already have used regular expressions. Remember `withoutPrefix.split(/ +/);`? This little `/ +/` is actually a Regular Expression. The `/` on either side tell JavaScript where the Regular Expression begins and where it ends, the stuff inbetween is it is content. 
+
+<tip>For a more detailed explanation of please consult the [MDN's documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp) about them.</tip>
+
+The RegEx you will use for user mentions will look like this: `/^<@!?(\d+)>$/`.
+Here is how the RegEx works:
+
+ 1. The `^` at the beginning and the `$` at the end mean that the mention has to take up the entire string.
+ 2. You have the typical `<@` and `>` at the beginning and end.
+ 3. The `?` after the `!` indicates that the `!` is optional.
+ 4. `\d+` means the RegEx will look for multiple digits, which will be the ID.
+ 5. The parentheses around the `\d+` create a capture group, which allows us to get the ID out of the mention.
+
+Using the `match` method on strings you can get the values of the capture group, i.e., the ID of the mention.
+
+<warning>
+Discord.js has [built-in patterns](https://discord.js.org/#/docs/main/stable/class/MessageMentions?scrollTo=s-CHANNELS_PATTERN)
+for matching mentions, however as of version 11.4 they do not contain any groups
+and thus aren't useful for actually getting the ID out of the mention.
+</warning>
+
+Updating your `getUserFromMention` function to use RegEx gives you this:
+
+```js
+function getUserFromMention(mention) {
+	const matches = mention.match(/^<@!?(\d+)>$/);
+	// The id is the first and only match found by the RegEx.
+	// However the first element in the matches array will be the entire mention, not just the ID,
+	// so use index 1.
+	const id = matches[1];
+	// Now just get the User with the ID and return that.
```

Comment:
I don't think this comment is entirely necessary, but the ones above it explaining the RegEx stuff are okay. 👍